sssd_test_framework/utils/sbus.py

```diff
@@ -0,0 +1,459 @@
+"""Manage D-Bus operations"""
+
+from __future__ import annotations
+
+import re
+import xml.etree.ElementTree as xml
+from abc import ABC
+from enum import Enum, auto
+from typing import Any
+
+from pytest_mh import MultihostHost, MultihostUtility
+from pytest_mh.ssh import SSHProcessResult
+
+__all__ = [
+    "SBUSUtils",
+    "DBUSBus",
+    "Variant",
+]
+
+
+class DBUSBus(Enum):
+    """
+    Types of predefined D-Bus buses.
+    """
+
+    SYSTEM = auto()
+    """The system bus"""
+    SESSION = auto()
+    """The session bus"""
+
+
+class Variant:
+    """
+    Python representation for the D-Bus ``variant`` type.
+
+    This is a container for other types.
+    """
+
+    def __init__(self, value):
+        self.value = value
+        """The actual value stored in the variant type"""
+
+
+class DBUSObject(MultihostUtility[MultihostHost]):
+    """
+    D-Bus access.
+
+    Objects of this class represent the root or child D-Bus nodes.
+    """
+
+    class ProxyAttribute(ABC):
```

Comment:
attribute kind of conflits with property, how about ProxyObject/ProxyBase?

Reply:
Changed it to ProxyObject.